src/ddprof_worker.cc

```diff
@@ -88,9 +88,22 @@ DDRes report_lost_events(DDProfContext &ctx) {
   return {};
 }
 
+int64_t get_perfclock_offset() {
+  auto system_now = std::chrono::system_clock::now();
+  auto perf_now = ddprof::PerfClock::now();
+
+  auto system_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                       system_now.time_since_epoch())
+                       .count();
+  auto perf_ns = perf_now.time_since_epoch().count(); // assumes offset is in ns
```

Comment:
You don't have to assume, `perf_now.time_since_epoch()` returns a duration and you can use c++ type system to ensure conversions are correct:
```
auto perf_clock_epoch = std::chrono::system_clock::now() - ddprof::PerfClock::now().time_since_epoch();
return std::chrono::nanoseconds(perf_clock_epoch.time_since_epoch()).count();
```

You could return a typed duration (std::chrono::nanoseconds) or even better a `std::chrono::system_clock::time_point` which would represent the epoch of `PerfClock` in `system_clock` referential.